File: 11/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height)-1
        max_cap = 0
        while left < right:
            max_cap = max(max_cap, min(height[left], height[right])*(right-left))
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        
        return max_cap

File: 11/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans = 0, len = height.size()-1;
        int left = 0;
        int right = len;
        int width = len;
        
        for(int i=0; i<len; i++){
            ans = max(ans, min(height[left], height[right])*width);
            if(height[left]>height[right])
                right--;
            else
                left++;
            width--;
        }
        return ans;
    }
};
